cpp/dataStructures/CMakeLists.txt

```diff
@@ -1,16 +1,12 @@
-cmake_minimum_required(VERSION 3.2.2)
-set(CMAKE_BUILD_TYPE Release)
-
-#For the shared library:
-set ( PROJECT_LINK_LIBS libcppCommon.dylib )
-link_directories( ../common/build )
+# However, the file(GLOB...) allows for wildcard additions:
+file(GLOB SOURCES "src/*.cpp")
 
-#Bring the headers, such as Student.h into the project
-include_directories(include)
-include_directories(../common/include/)
+# Generate the shared library from the sources
+add_library(cppDS SHARED ${SOURCES})
 
```

Comment:
It has to be this way round since it's `target_include_directories` as opposed to the general `include_directories`; i.e. it applies the include dirs to a single, specific CMake target.  This requires the target to have been defined first (usually via `add_library` or `add_executable`).